File: tests/fixtures/multi-language/sample.go
// Go 测试文件
package main

import (
	"errors"
	"regexp"
	"sync"
)

type User struct {
	ID    string
	Name  string
	Email string
}

type UserService struct {
	users map[string]*User
	mu    sync.RWMutex
}

func NewUserService() *UserService {
	return &UserService{
		users: make(map[string]*User),
	}
}

func (s *UserService) GetUser(id string) (*User, error) {
	s.mu.RLock()
	defer s.mu.RUnlock()
	
	user, ok := s.users[id]
	if !ok {
		return nil, errors.New("user not found")
	}
	return user, nil
}

func (s *UserService) CreateUser(user *User) error {
	s.mu.Lock()
	defer s.mu.Unlock()
	
	s.users[user.ID] = user
	return nil
}

func (s *UserService) DeleteUser(id string) bool {
	s.mu.Lock()
	defer s.mu.Unlock()
	
	_, exists := s.users[id]
	if exists {
		delete(s.users, id)
	}
	return exists
}

func ValidateEmail(email string) bool {
	pattern := `^[^\s@]+@[^\s@]+\.[^\s@]+$`
	matched, _ := regexp.MatchString(pattern, email)
	return matched
}

File: tests/fixtures/multi-language/sample.ts
// TypeScript 测试文件

export interface User {
  id: string;
  name: string;
  email: string;
}

export class UserService {
  private users: Map<string, User> = new Map();

  async getUser(id: string): Promise<User | null> {
    return this.users.get(id) || null;
  }

  async createUser(user: User): Promise<void> {
    this.users.set(user.id, user);
  }

  async deleteUser(id: string): Promise<boolean> {
    return this.users.delete(id);
  }
}

export function validateEmail(email: string): boolean {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return regex.test(email);
}
